.github/workflows/ci.yml

```
@@ -61,10 +55,65 @@ jobs:
 
       - name: Print the installed version
         run: echo "Installed uv version is ${{ steps.setup-uv.outputs.uv-version }}"
-
       - name: Install dependencies
         run: uv sync --frozen --group devel
-
       - name: Run tests
         run: |
            uv run --frozen pytest -vv
+
+  test-macos:
+    name: macos-latest
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      
+      # Set up Python 3.12 on macOS environment
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      
+      # Install uv and project dependencies inside a virtual environment
+      - name: Install uv and dependencies
+        run: |
+          python3 -m venv .venv
+          source .venv/bin/activate
+          pip install uv
+          uv pip install --editable . --group devel
+      
+      # Install macOS-specific external tools (such as XML parsers and Java)
+      - name: Install external tools (macOS)
+        run: |
+          brew install libxml2 libxslt openjdk
+
+          # Define the version and URL for jing.jar
+          JING_VERSION="20091111"
+          JING_URL="https://repo1.maven.org/maven2/com/thaiopensource/jing/20091111/jing-20091111.jar"
+
+          # Download jing.jar file for XML validation
+          curl -sL -f $JING_URL -o jing.jar
+          
+          # Ensures the jar file is successfully downloaded, else exit with error
+          if [ ! -f jing.jar ]; then
+              echo "Error: Failed to download jing.jar"
```

I'd recommend to use the error message from the [GitHub docs](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-commands#example-creating-an-annotation-for-an-error):

```suggestion
              echo "::error file=$JING_URL:: Failed to download jing.jar"
```

This will print the error in red.

---

Modified, thanks.